corehq/apps/case_search/filter_dsl.py

```diff
@@ -160,54 +157,36 @@ def _is_subcase_count(node):
 
         return isinstance(node.left, FunctionCall) and node.left.name == 'subcase-count'
 
-    def _raise_step_RHS(node):
-        raise CaseFilterError(
-            _("You cannot reference a case property on the right side "
-              "of a boolean operation. If \"{}\" is meant to be a value, please surround it with "
-              "quotation marks").format(serialize(node.right)),
-            serialize(node)
-        )
-
-    def _equality(node):
-        """Returns the filter for an equality operation (=, !=)
-
-        """
-        acceptable_rhs_types = (int, str, float, FunctionCall, UnaryExpression)
-        if isinstance(node.left, Step) and (
-                isinstance(node.right, acceptable_rhs_types)):
-            # This is a leaf node
-            case_property_name = serialize(node.left)
-            value = unwrap_value(node.right, context)
-            q = case_property_query(case_property_name, value, fuzzy=context.fuzzy)
-
-            if node.op == '!=':
-                return filters.NOT(q)
-
-            return q
-
-        if isinstance(node.right, Step):
-            _raise_step_RHS(node)
-
-        raise CaseFilterError(
-            _("We didn't understand what you were trying to do with {}").format(serialize(node)),
-            serialize(node)
-        )
-
     def _comparison(node):
-        """Returns the filter for a comparison operation (>, <, >=, <=)
+        """Returns the filter for a comparison operation (=, !=, >, <, >=, <=)
 
         """
-        try:
-            case_property_name = serialize(node.left)
-            value = unwrap_value(node.right, context)
-            return case_property_range_query(case_property_name, **{COMPARISON_MAPPING[node.op]: value})
-        except (TypeError, ValueError):
+        return _comparison_raw(node.left, node.op, node.right, node)
+
+    def _comparison_raw(case_property_name_raw, op, value_raw, node):
+        if not isinstance(case_property_name_raw, Step):
             raise CaseFilterError(
-                _("The right hand side of a comparison must be a number or date. "
-                  "Dates must be surrounded in quotation marks"),
-                serialize(node),
+                _("We didn't understand what you were trying to do with {}").format(serialize(node)),
+                serialize(node)
             )
 
+        case_property_name = serialize(case_property_name_raw)
+        value = unwrap_value(value_raw, context)
+        if op in [EQ, NEQ]:
+            query = case_property_query(case_property_name, value, fuzzy=context.fuzzy)
+            if op == NEQ:
+                query = filters.NOT(query)
+            return query
+        else:
+            try:
+                return case_property_range_query(case_property_name, **{RANGE_OP_MAPPING[op]: value})
+            except (TypeError, ValueError):
```

Comment:
Note: you're probably already aware, but this `except` no longer is active on these lines:

```
case_property_name = serialize(case_property_name_raw)
value = unwrap_value(value_raw, context)
```

While the error handling you've added to `unwrap_value` looks good, this more generic `except` catches a wider range of errors so on HQ an error previously caught by the `except` might not be anymore. I don't think they're important to except though? So just a warning from me

Reply:
I was just reading this line too.  It looks like this is intended to catch the `TypeError` raised here:
https://github.com/dimagi/commcare-hq/blob/a7b5a7fa10280c45f7cc46ac73046d0356fc26c4/corehq/apps/es/case_search.py#L221-L222
It does seem like the type of thing the DSL parser should be catching, rather than `ESQuery`.  I think it's fine to say that's out of the scope of this refactor, though.

Reply:
I did specifically move those out of the try catch because they should not be raising errors and if they do we want it to fail loudly so that we can handle them better.